Feat: Settings page Ui has been created

drakkomaximo opened the pull request:
close #72 

# Settings Page Implementation

## Overview
Implemented a new settings page for the dashboard that allows users to manage their preferences and notification settings. The page follows the existing dashboard design language and provides a seamless user experience.

## Features

### User Interface
- Modern, responsive design using Shadcn UI components
- Dark theme consistent with the dashboard
- Tab-based navigation for different settings sections
- Form validation and error handling
- Loading states and feedback notifications

### Functionality
- **Preferences Management**
  - Language selection
  - Timezone configuration
  - Theme preferences
- **Notification Settings**
  - Email notifications toggle
  - Push notifications toggle
  - Project updates notifications toggle

### Technical Implementation
- Built with React and Next.js
- Form handling using react-hook-form
- Validation using Zod schema
- State persistence using localStorage
- Toast notifications using react-hot-toast
- Responsive design with Tailwind CSS

## Technical Details

### Form Validation
```typescript
const settingsSchema = z.object({
  language: z.string().min(1, "Language is required"),
  timezone: z.string().min(1, "Timezone is required"),
  theme: z.string().min(1, "Theme is required"),
  emailNotifications: z.boolean(),
  pushNotifications: z.boolean(),
  projectUpdates: z.boolean(),
});
```

### State Management
- Form state managed by react-hook-form
- Settings persisted in localStorage
- Real-time validation feedback
- Loading states for async operations

### UI Components
- Tabs for section navigation
- Cards for content organization
- Input fields with validation
- Checkboxes for toggle options
- Buttons with loading states

## Accessibility
- Semantic HTML structure
- Proper ARIA labels
- Keyboard navigation support
- Color contrast compliance
- Responsive design for all screen sizes

## Future Improvements
- Integration with backend API
- Additional settings sections
- Theme customization options
- User profile management
- Advanced notification settings

![image](https://github.com/user-attachments/assets/d6db42e3-55a7-4bcf-b9ca-203f1e07136d)
![image](https://github.com/user-attachments/assets/ad5d7455-185b-44bb-b056-0c2c8a18c9ff)

Pvsaint commented:
@drakkomaximo fix the width of your component also, the navbar is looking compressed

drakkomaximo commented:
# Settings Page Implementation

## Overview
This document outlines the implementation of the settings page and its components, as well as current known issues.

## Implementation Details

### Settings Form Component
- Created a new `SettingsForm` component in `src/components/settings/SettingsForm.tsx`
- Implemented form handling using `react-hook-form`
- Added form validation using `zod` schema
- Implemented state persistence using `localStorage`
- Created a responsive layout with tabs for different settings sections

### Form Fields
1. **Preferences Tab**
   - Language selection (dropdown)
   - Timezone selection (dropdown)
   - Theme selection (dropdown)

2. **Notifications Tab**
   - Email notifications toggle
   - Push notifications toggle
   - Project updates toggle

### Technical Implementation
- Used Shadcn UI components for consistent styling
- Implemented dark theme support
- Added form validation and error handling
- Added loading states and feedback notifications
- Implemented responsive design with Tailwind CSS

## Current Issues

### Navbar Responsiveness
The navbar currently has the following issues:
- Mobile menu positioning problems
- Inconsistent behavior on different screen sizes
- Menu items not properly aligned in mobile view
- Hamburger menu button positioning issues

**Note**: These issues are not addressed in this PR as they are outside the current scope. A separate task should be created to handle navbar responsiveness improvements.

## Important Notes

### Selector Values
The values used in the selectors (language, timezone) are illustrative examples and should be replaced with actual available options from the backend or configuration. The only exception is the theme selector, which uses standard values:
- `dark`
- `light`
- `system`

### Future Improvements
1. Integration with backend API for actual settings management
2. Implementation of real-time theme switching
3. Addition of more settings sections as needed
4. Improved error handling and validation
5. Addition of user profile management features

## Testing
The implementation has been tested for:
- Form validation
- State persistence
- Responsive layout
- Theme consistency
- Error handling

## Dependencies
- react-hook-form
- zod
- @hookform/resolvers/zod
- react-hot-toast
- Shadcn UI components 


https://github.com/user-attachments/assets/2e85e6a6-de19-4434-bf98-7634f67f50c6

Pvsaint commented:
@drakkomaximo Please your PR message from the ReadMe file before I merge.
